scrapy/squeues.py

```diff
@@ -85,26 +86,47 @@ def _pickle_serialize(obj):
     # Both pickle.PicklingError and AttributeError can be raised by pickle.dump(s)
     # TypeError is raised from parsel.Selector
     except (pickle.PicklingError, AttributeError, TypeError) as e:
-        raise ValueError(str(e)) from e
+        raise SerializationError(str(e)) from e
+
+
+def _ignore_args_kwargs_passed_to_constructor(queue_class, used_kwargs):
+    class AcceptingQueue(queue_class):
+        def __init__(self, path, *_, **kwargs):
+            new_kwargs = {k: kwargs[k] for k in used_kwargs if k in kwargs}
+            super().__init__(path, **new_kwargs)
```

Comment:
Won’t this cause `chunksize` to now get ignored if it used to be based as a positional argument instead of a keyword argument?

Reply:
Yes, you are right. But this very complex question. Single place where `chanksize=smth` is used in scrapy is in `tests/test_squeue.py`. It is used as named argument. Chunking behaviour tested there is not an attribute of scrapy and tested in `queuelib`. My proposition is to remove these chnunking tests. After (or in this PR) we can present something like `SCHEDULER_DISK_QUEUE_CHUNKSIZE` into scrapy's settings.

Reply:
My worry is backward compatibility.

Ideally, `chunksize` should continue to work as a positional argument, and if we want that to stop working in the future, we should emit the corresponding warning.

If we consider it extremely unlikely that this backward-incompatible change will affect anyone, we can simply acknowledge it in the release notes and go ahead. But we need to remember to cover it in the release notes.

Reply:
> Ideally, chunksize should continue to work as a positional argument, and if we want that to stop working in the future

There was no places in scrapy where this argument was used. Nothing to continue heree

Reply:
Hey @Gallaecio, implementation is provided  to take `chunksize` from `crawle.settings` (without docs etc.). Just to check validity of idea.

Reply:
I’m not against the idea, but if you are suggesting that as a solution for the backward-incompatibility issue, I _think_ (my memory about this patch has almost faded) it does not address it (we still had a `parameter` which is not expected anymore, so if used things will fail).

However, as [I said before](https://github.com/scrapy/scrapy/pull/4783#discussion_r511587394), I don’t mind if we do make the backward-compatibility change that is unlikely to break any existing code, provided we do mention it in the release notes just in case.

Reply:
> Does MarshalFifoDiskQueueNonRequest.__init__ accept a chunksize parameter after the last change?

No, now they accept `crawler` argument (as other scrapy objects) and read `chunksize` from `crawler.settings['SCHEDULER_DISK_QUEUE_CHUNKSIZE']`

>  provided we do mention it in the release notes just in case.

Should I create a section for future release in `docs/news.rst`?

Reply:
> Should I create a section for future release in docs/news.rst?

While we usually look for the backward-incompatible label when covering issues in the release notes, it may be good to do that to make sure we do not forget. You don’t even need to work to hard on in, we just need to make sure that, when updating the release notes, we do not forget about covering this.

Reply:
Resolved in https://github.com/scrapy/scrapy/pull/4783#discussion_r619700203